spm.py

```diff
@@ -1,24 +1,379 @@
+#   Cathode
+Vol_ca = (4*np.pi/3)*r_ca**3 # Volume of a single anode particle [m^3]
+A_surf_ca = 4*np.pi*r_ca**2 # geometric surface area of a single anode particle [m^2]
+A_s_ca = 3/r_ca # ratio of surface area to volume for the graphite anode [1/m]
+A_sg_ca = Epsilon_g*Delta_y_ca*A_s_ca # interface surface area per geometric surface area [m^2_interface/m^2_geometric]
+
+# Initial Concentrations
+# The number of moles of C6 in a single anode particle [mol]. Divide by 6 to 
+#    convert MW from per mole of carbon to per mole of C6 [mol]
+C_g = rho_g/MW_g/6 # Molar concentration of C6 in the anode [mol/m^3]
+
+X_g_0_an = 1 - X_Li_0_an # Initial Mole Fraction of Graphite [-]
+C_Li_0_an = C_g*X_Li_0_an # Initial Lithium molar concentration in the Anode [mol_Li/m^3-graphite]
+C_g_0_an = C_g*X_g_0_an # Initial graphite molar concentration in the Anode [mol_Li/m^3-graphite]
+
+C_FePO4 = rho_FePO4/MW_FePO4 # Molar concentration of FePO4 in the cathode [mol/m^3]
+
+X_FePO4_0 = 1 - X_Li_0_ca # Initial Mole Fraction of FePO4 [-]
+C_Li_0_ca = C_FePO4*X_Li_0_ca # Initial Lithium molar concentration in the Anode [mol_Li/m^3-FePO4]
+C_FePO4_0 = C_FePO4*X_FePO4_0 # Initial FePO4 molar concentration in the Anode [mol_Li/m^3-FePO4]
+
+'''
+Set up the Half Cells
+'''
+# The initial concentrations of LiC6 and C6 depend on the user defined initial mole fractions, 
+#   the concentraions of Li+ in the electroltyes are static and the same in both half cells
+# I am not confident in my standard entropy values for the cathode, but this is not an issue
+#   yet because the cell is still at standard temperature
+
+# Anode reacation: LiC6 -> C6 + Li+ + e-
+LiC6 = Species("LiC6",-230000,-11.2,C_g,0)
+C6 = Species("C",0,0,C_g,0)
+Li_plus = Species("Li+",-293300,49.7,C_std,1)
+
+LiC6_rxn_an = Participant(LiC6,1,C_Li_0_an,gamma_LiC6)
+C6_rxn_an = Participant(C6,1,C_g_0_an,gamma_C6)
+Li_plus_rxn_an = Participant(Li_plus,1,C_Li_plus,gamma_Li_plus)
+
+React_an = [LiC6_rxn_an]
+Prod_an = [Li_plus_rxn_an,C6_rxn_an]
+
+# Both electrodes have the same reation and in this forward reaction one electron is 
+#   produced so n = 1 (as it is defined above in the inputs)
+Anode = Half_Cell(React_an,Prod_an,n,T,Beta,F,R,Cap_dl,i_o_reff,A_sg_an,A_s_an,'LiC6','Li+')
+
+# Cathode reacation: LiFePO4 -> FePO4 + Li+ + e-
+LiFePO4 = Species("LiFePO4",-326650,130.95,C_FePO4,0)
+FePO4 = Species("FePO4",0,171.3,C_FePO4,0)
+
+LiFePO4_rxn_ca = Participant(LiFePO4,1,C_Li_0_ca,gamma_LiFePO4)
+FePO4_rxn_ca = Participant(FePO4,1,C_FePO4_0,gamma_FePO4)
+Li_plus_rxn_ca = Participant(Li_plus,1,C_Li_plus,gamma_Li_plus)
+
+React_ca = [LiFePO4_rxn_ca]
+Prod_ca = [Li_plus_rxn_ca,FePO4_rxn_ca]
+
+Cathode = Half_Cell(React_ca,Prod_ca,n,T,Beta,F,R,Cap_dl,i_o_reff,A_sg_ca,A_s_ca,'LiFePO4','Li+')
+
+'''
+Integration
+'''
+# Integration Limits
+def min_voltage(_,SV,i_ext,Anode,Cathode):
+    V_cell = SV[2] + i_ext*t_sep/sigma_sep - SV[0]
```

Comment:
I think the sign on the `i_ext` term should be negative, here and on line 162.

This makes sense from theory (V_cell drops with increasing discharge current, which here is a positive current), and is consistent with what you have on lines 255-265, below.

Reply:
yes, I think this is a mistake from when I had my entire sign convention backwards and I did not update it here

Reply:
Oh, I see - I had missed that!  

Okay, looks good to me - merging now.